### PR TITLE
Update/simplify code for the new coarser groups paradigm

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description

In this pull request I update/simplify the Terraform code for the new coarser groups paradigm.

## 💭 Motivation and Context

This change from the fine-grained groups we have been creating to coarser groups that are defined by job function is necessary because AWS has a hard limit of ten for the number of groups to which a user can belong.

## 🧪 Testing

I deployed these changes to production, and I verified that all pre-commit hooks pass.

## 🚥 Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jsf9k** [comment]
When reviewing, I suggest you start by looking at the changes in the `provisionaccount-role-tf-module` and `users` subdirectories.  Once you understand those changes, the changes in the other subdirectories are pretty straightforward.

File: users/gods_group_membership.tf [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Put our admin IAM users in the Terraform backend access and user
```
Comment (dav3r) [inline users/gods_group_membership.tf]: You missed changing this "admin" to "god-like".

File: provisionaccount-role-tf-module/README.md [hunk 2]
```diff
@@ -46,23 +36,17 @@ module "provisionaccount" {
...
-| assume_provisionaccount_policy_name | The name to associate with the IAM policy that allows assumption of the role with sufficient permissions to provision all AWS resources in the new account (e.g. "DNS-AssumeProvisionAccount"). | string | | yes |
 | aws_region | The AWS region where the non-global resources for this account are to be created (e.g. us-east-1). | string | `us-east-1` | no |
 | new_account_id | The ID of the account being configured. | string | | yes |
-| provisionaccount_role_description | The description to associate with the IAM role (as well as the corresponding policy) that allows sufficient access to provision all AWS resources in the new account (e.g. \Allows sufficient permissions to provision all AWS resources in the DNS account."). | string | | yes |
+| provisionaccount_role_description | The description to associate with the IAM role (as well as the corresponding policy) that allows sufficient access to provision all AWS resources in the new account (e.g. "Allows sufficient permissions to provision all AWS resources in the DNS account."). | string | | yes |
```
[the remark this answers is not in the document]
Comment (jsf9k) [reply provisionaccount-role-tf-module/README.md]: Those DNS mentions are exempli gratia, so they should be specific.

File: users/gods_group_membership.tf (hunk 1 above)
Comment (jsf9k) [reply users/gods_group_membership.tf]: Remedied in 8397be1.

**Pull request merged**